addons/mail/models/discuss/discuss_channel.py

```diff
@@ -824,7 +824,8 @@ def _find_or_create_persona_for_channel(self, guest_name, timezone, country_code
                 ).sudo(False)
                 guest._set_auth_cookie()
                 self = self.with_context(guest=guest)
-            self.add_members(guest_ids=guest.ids, post_joined_message=post_joined_message)
+            if self.channel_type != 'channel':
```

Comment:
This condition isn't complete. It would not work for channels of other types ex. Group. So, instead we should pass a variable to determine whether to add guest or not.